likecoin/admin/views/matters.php

```diff
@@ -134,8 +171,26 @@ function likecoin_replace_matters_attachment_url( $content, $params ) {
 		}
 		if ( ! $attachment_id && $url ) {
 			$attachment_id = attachment_url_to_postid( $url );
+			// if its url image.
+			$image_infos = get_post_meta( $post_id, LC_IMAGE_INFO, true );
+			if ( $image_infos ) {
+				foreach ( $image_infos as $keys => $values ) {
+					if ( is_array( $values ) || is_object( $values ) ) {
+						$matters_image_url                       = $values->matters_url;
+						$matters_url_components                  = preg_split( '#/#', $matters_image_url );
+						$attachment_id_with_mime_type            = end( $matters_url_components );
+						$attachment_id_with_mime_type_components = preg_split( '#[.]#', $attachment_id_with_mime_type );
+						$attachment_id_without_mime_type         = reset( $attachment_id_with_mime_type_components );
+						if ( $values->original_url === $url ) {
+							$image->setAttribute( 'src', $matters_image_url );
+							$image->setAttribute( 'data-asset-id', $attachment_id_without_mime_type );
+						}
+					}
```

Comment:
just save the `attachment_id` into metadata when uploading the image, saves a lot of complex code
Also should use a map for $image_infos, currently we are looping through every $image_info once for each image in the post, which is not ideal and meaningless